src/compiler/core.ts

```diff
@@ -124,6 +124,13 @@ namespace ts {
         return undefined;
     }
 
+    export function zip<T, U>(arrayA: T[], arrayB: U[], callback: (a: T, b: U, index: number) => void): void {

```

Comment:
lodash and ramda call this `zipWith`, and have `zip` with the type `(t: T[], u: U[]) => [T,U][]` (as do underscore, F#, Python and Haskell). I'd call it `zipWith`